plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeConnectorTest.java

```diff
@@ -1820,6 +1820,65 @@ public void testTableWithTrailingSlashLocation(boolean partitioned)
+        assertUpdate(format("DELETE FROM %s WHERE %s", table, deleteFilter), 2);
+        assertQuery("SELECT customer, purchases, address FROM " + table, "VALUES ('Mary', 10, 'Adelphi'), ('Aaron', 3, 'Dallas')");
+        assertUpdate("DROP TABLE " + table);
+    }
+
+    @DataProvider
+    public Object[][] deleteFiltersForTable()
+    {
+        return new Object[][]{
+                {
+                        "CREATE TABLE %s (customer VARCHAR, purchases INT, address VARCHAR) WITH (location = 's3://%s/%s')",
+                        "address = 'Antioch'"
+                },
+                {
+                        // delete filter applied on function over non-partitioned field
+                        "CREATE TABLE %s (customer VARCHAR, address VARCHAR, purchases INT) WITH (location = 's3://%s/%s', partitioned_by = ARRAY['address'])",
+                        "starts_with(address, 'Antioch')"
+                },
+                {
+                        // delete filter applied on partitioned field
+                        "CREATE TABLE %s (customer VARCHAR, address VARCHAR, purchases INT) WITH (location = 's3://%s/%s', partitioned_by = ARRAY['address'])",
+                        "address = 'Antioch'"
+                },
+                {
+                        // delete filter applied on function over partitioned field
+                        "CREATE TABLE %s (customer VARCHAR, address VARCHAR, purchases INT) WITH (location = 's3://%s/%s', partitioned_by = ARRAY['address'])",
+                        "starts_with(address, 'Antioch')"
+                },
+                {
+                        // delete filter applied on non-partitioned field
+                        "CREATE TABLE %s (customer VARCHAR, address VARCHAR, purchases INT) WITH (location = 's3://%s/%s', partitioned_by = ARRAY['customer'])",
+                        "address = 'Antioch'"
+                },
+                {
+                        // delete filter fully applied on composed partitioned
+                        "CREATE TABLE %s (purchases INT, customer VARCHAR, address VARCHAR) WITH (location = 's3://%s/%s', partitioned_by = ARRAY['address', 'customer'])",
+                        "address = 'Antioch' AND (customer = 'Aaron' OR customer = 'Bill')"
+                },
+                {
+                        // delete filter applied only partly on first partitioned field
+                        "CREATE TABLE %s (purchases INT, address VARCHAR, customer VARCHAR) WITH (location = 's3://%s/%s', partitioned_by = ARRAY['address', 'customer'])",
+                        "address = 'Antioch'"
+                },
+                {
+                        // delete filter applied only partly on second partitioned field
+                        "CREATE TABLE %s (purchases INT, address VARCHAR, customer VARCHAR) WITH (location = 's3://%s/%s', partitioned_by = ARRAY['customer', 'address'])",
+                        "address = 'Antioch'"
+                }
```

Comment:
Extra `,` will reduce the diff when adding the new condition in the future:
```suggestion
                },
```